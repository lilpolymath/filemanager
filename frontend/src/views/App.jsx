import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useInterval } from 'use-interval';

import FileExplorer from '../components/FileExplorer';
import FileContent from '../components/FileContent';
import { refreshFiles } from '../store/files/files-actions';

import styles from "./App.module.css";

const POLL_INTERVAL = 5000;

function App() {
  const dispatch = useDispatch();

  useInterval(() => {
    dispatch(refreshFiles());
  }, POLL_INTERVAL, true);

  const { list, hasError } = useSelector(store => store.files.file);
  const { selectedFile } = useSelector(store => store.filemanager);

  return (
    <div className={styles.AppContainer}>
      {
        hasError ? <div className={styles.Error}>
          <h3>An error occured while fetching the files</h3>
        </div> : (
          <div className={styles.App}>
            <FileExplorer files={list} />
            <FileContent file={selectedFile} />
          </div>
        )
      }
    </div>
  );
}

export default App;
