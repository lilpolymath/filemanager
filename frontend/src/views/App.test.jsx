import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import App from './App';

describe('App Component', () => {
  it('should render defaults', () => {
    const initialState = {
      files: {
        file: {
          list: [],
          isLoading: true
        }
      },
      filemanager: {
        selectedFile: null
      }
    };
    const store = configureMockStore([thunk])(initialState);
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
