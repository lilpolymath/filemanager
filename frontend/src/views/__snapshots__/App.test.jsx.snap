// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render defaults 1`] = `
<div
  class="AppContainer"
>
  <div
    class="App"
  >
    <section
      class="FileExplorer"
    >
      <div
        class="FileExplorerEmpty"
      >
        The root folder is empty.
      </div>
    </section>
    <section
      class="FileContent"
    >
      <pre />
    </section>
  </div>
</div>
`;
